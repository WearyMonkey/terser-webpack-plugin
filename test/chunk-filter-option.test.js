import path from "path";

import TerserPlugin from "../src/index";

import {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  readsAssets,
} from "./helpers";

describe("chunk filter option", () => {
  let compiler;

  beforeEach(() => {
    compiler = getCompiler({
      entry: {
        include: path.resolve(__dirname, "./fixtures/entry.js"),
        dontInclude: path.resolve(__dirname, "./fixtures/file.js"),
      },
      output: {
        path: path.resolve(__dirname, "./dist"),
        filename: `[name].js`,
        chunkFilename: `[id].[name].js`,
      },
    });
  });

  it("should match snapshot with empty value", async () => {
    new TerserPlugin().apply(compiler);

    const stats = await compile(compiler);

    expect(readsAssets(compiler, stats)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should match snapshot for a chunk filter that only includes one chunk", async () => {
    const chunkFilter = jest.fn((chunk) => chunk.name === "include");
    new TerserPlugin({
      chunkFilter: (chunk, name) => chunkFilter(chunk, name),
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(chunkFilter).toHaveBeenCalledWith(
      expect.objectContaining({ name: "include" }),
      "include.js"
    );
    expect(chunkFilter).toHaveBeenCalledWith(
      expect.objectContaining({ name: "dontInclude" }),
      "dontInclude.js"
    );
    expect(readsAssets(compiler, stats)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });
});
