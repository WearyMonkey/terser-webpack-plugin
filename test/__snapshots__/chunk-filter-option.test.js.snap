// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chunk filter option should match snapshot for a chunk filter that only includes one chunk: assets 1`] = `
Object {
  "dontInclude.js": "/******/ (() => { // webpackBootstrap
/******/ 	var __webpack_modules__ = ({

/***/ 993:
/***/ ((module) => {

module.exports = \\"ok\\";


/***/ })

/******/ 	});
/************************************************************************/
/******/ 	// The module cache
/******/ 	var __webpack_module_cache__ = {};
/******/ 	
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/ 		// Check if module is in cache
/******/ 		if(__webpack_module_cache__[moduleId]) {
/******/ 			return __webpack_module_cache__[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = __webpack_module_cache__[moduleId] = {
/******/ 			// no module.id needed
/******/ 			// no module.loaded needed
/******/ 			exports: {}
/******/ 		};
/******/ 	
/******/ 		// Execute the module function
/******/ 		__webpack_modules__[moduleId](module, module.exports, __webpack_require__);
/******/ 	
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/ 	
/************************************************************************/
/******/ 	// startup
/******/ 	// Load entry module
/******/ 	// This entry module is referenced by other modules so it can't be inlined
/******/ 	__webpack_require__(993);
/******/ })()
;",
  "include.js": "(()=>{var r={791:r=>{r.exports=function(){console.log(7)}}},o={};!function t(e){if(o[e])return o[e].exports;var n=o[e]={exports:{}};return r[e](n,n.exports,t),n.exports}(791)})();",
}
`;

exports[`chunk filter option should match snapshot for a chunk filter that only includes one chunk: errors 1`] = `Array []`;

exports[`chunk filter option should match snapshot for a chunk filter that only includes one chunk: warnings 1`] = `Array []`;

exports[`chunk filter option should match snapshot with empty value: assets 1`] = `
Object {
  "dontInclude.js": "(()=>{var r={993:r=>{r.exports=\\"ok\\"}},t={};!function e(o){if(t[o])return t[o].exports;var p=t[o]={exports:{}};return r[o](p,p.exports,e),p.exports}(993)})();",
  "include.js": "(()=>{var r={791:r=>{r.exports=function(){console.log(7)}}},o={};!function t(e){if(o[e])return o[e].exports;var n=o[e]={exports:{}};return r[e](n,n.exports,t),n.exports}(791)})();",
}
`;

exports[`chunk filter option should match snapshot with empty value: errors 1`] = `Array []`;

exports[`chunk filter option should match snapshot with empty value: warnings 1`] = `Array []`;
